Source/SimpleActions/Private/SimpleAction.cpp

```diff
@@ -135,17 +148,51 @@ void USimpleAction::StartAction(UPARAM(DisplayName = "ActionActor") AActor* NewA
     }
 
     UE_LOG(LogSimpleAction, Verbose, TEXT("StartAction: Starting. Action '%s'."), *GetName());
-    ActionStart();
+
     OnActionStarted.Broadcast(this);
+
+    if(DurationOverride > 0.f) {
+	    const FTimerDelegate DurationOverrideDelegate = FTimerDelegate::CreateUObject(this, &USimpleAction::FinishAction, true);
+        GetWorld()->GetTimerManager().SetTimer(Timer_DurationOverride, DurationOverrideDelegate, DurationOverride + StartDelay, false);
+    }
+
+    if(StartDelay <= 0) {
+		ActionStart(); 
+    } else {
+        const FTimerDelegate StartDelayDelegate = FTimerDelegate::CreateUObject(this, &USimpleAction::ActionStart);
+        GetWorld()->GetTimerManager().SetTimer(Timer_StartDelay, StartDelayDelegate, StartDelay, false);
+    }
+}
+
+void USimpleAction::StartAction_Bound(AActor* NewActionActor, AActor* NewInstigator, FOnActionStartedDelegate Started, FOnActionInterruptedDelegate Interrupted, FOnActionEndedDelegate Ended, FOnActionCompleteDelegate Complete, FOnActionFailedDelegate Failed) {
```

Comment:
I'm iffy about this pattern being in here.

Reply:
If you can figure out another method I'm all ears.